chain/src/sync/state_payload.rs

```diff
@@ -17,20 +17,26 @@ pub enum StatePayload {
         source: NoteSource,
         note: EncryptedNote,
     },
+    Swap {
+        source: NoteSource,
+        swap: SwapPayload,
+    },
```

Comment:
We add a new `Swap` variant, which also means that the `SwapPayload`s (encrypted `SwapPlaintext`s) go in the compact block and don't need to be downloaded seperately.